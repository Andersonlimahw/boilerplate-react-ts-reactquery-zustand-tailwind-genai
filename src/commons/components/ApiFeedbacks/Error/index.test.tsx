import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { ErrorApiComponent } from "./index";

describe("ErrorApiComponent", () => {
  const onRetryMock = vi.fn();

  it("deve renderizar o componente corretamente", () => {
    const { container } = render(<ErrorApiComponent onRetry={onRetryMock} />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("deve conter os elementos visuais esperados", () => {
    render(<ErrorApiComponent onRetry={onRetryMock} />);
    expect(screen.getByText("Sorry!")).toBeInTheDocument();
    expect(screen.getByTestId("error-api-component")).toBeInTheDocument();
  });
});
