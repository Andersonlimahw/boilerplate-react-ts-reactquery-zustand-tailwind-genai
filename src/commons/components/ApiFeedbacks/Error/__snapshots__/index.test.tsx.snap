// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorApiComponent > deve renderizar o componente corretamente 1`] = `
<div>
  <div
    class="rounded overflow-hidden shadow-lg"
    data-testid="error-api-component"
  >
    <div
      class="px-6 py-4"
    >
      <div
        class="font-bold text-xl mb-2"
      >
        Sorry!
      </div>
      <p
        class="text-gray-700 text-base"
      >
        A error happens D:
      </p>
    </div>
    <div
      class="px-6 pt-4 pb-2"
    >
      <span
        class="button inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2 cursor-pointer"
      >
         
        try again
         
      </span>
    </div>
  </div>
</div>
`;
